pkg/adapter/nimbus-kyverno/processor/kpbuilder.go

```diff
@@ -67,12 +96,76 @@ func cocoRuntimeAddition(np *v1alpha1.NimbusPolicy, rule v1alpha1.Rule) kyvernov
 	}
 	patchBytes, err := json.Marshal(patchStrategicMerge)
 	if err != nil {
-		panic(err)
+		errs = append(errs, err)
 	}
 	if err != nil {
-		panic(err)
+		errs = append(errs, err)
 	}
-	kp := kyvernov1.Policy{
+
+	deploymentsGVR := schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments"}
+	deployments, err := client.Resource(deploymentsGVR).Namespace(np.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	for _, d := range deployments.Items {
+		for key1, value1 := range d.GetLabels() {
+			for key2, value2 := range np.Spec.Selector.MatchLabels {
+				if key1 == key2 && value1 == value2 {
+					deployNames = append(deployNames, d.GetName())
+				}
+			}
+		}
+	}
+
+	for _, deployName := range deployNames {
+		mutateResourceSpec := kyvernov1.TargetResourceSpec{
+			ResourceSpec: kyvernov1.ResourceSpec{
+				APIVersion: "apps/v1",
+				Kind:       "Deployment",
+				Name:       deployName,
+			},
+		}
+		mutateTargetResourceSpecs = append(mutateTargetResourceSpecs, mutateResourceSpec)
+	}
+	if len(labels) > 0 {
+		for key, value := range labels {
+			resourceFilter := kyvernov1.ResourceFilter {
+				ResourceDescription: kyvernov1.ResourceDescription{
+					Kinds: []string{
+						"apps/v1/Deployment",
+					},
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							key: value,
+						},
+					},
+				},
+			}
+			matchResourceFilters = append(matchResourceFilters, resourceFilter)
+		}
+	} else if len(labels) == 0 {
+		mutateResourceSpec := kyvernov1.TargetResourceSpec{
```

Comment:
Why is this case needed ? You have already evaluated the list of deploys which are to be modified. If the len of labels is zero, what is this mutation target ?

Reply:
lets say if we have a set of deployments existing in the namespace and we want to mutate all those deployments, so we can omit the labels, in the TargetResourceSpec and apply the mutation to all the deploys in that namespace. @shivaccuknox

Reply:
The mutateTargetResources already consists the list of all deploys in that namespace. Looks like there is no need to append another mutateResourceSpec to it.